File: app/src/main/java/com/erotc/learning/util/ApplicationUtil.java
package com.erotc.learning.util;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.erotc.learning.R;
import com.erotc.learning.repository.DictionaryRepository;

/**
 * Created on 11/12/2018.
 */
public class ApplicationUtil {
    private static final String LOG_TAG = ApplicationUtil.class.getSimpleName();

    public static boolean shouldInitialize(Context context){
        DictionaryRepository repository = DictionaryRepository.getInstance(context);
        return ( repository.isDictionaryEmpty() || repository.isQuestionSetEmpty() );
    }

    public static View inflateButton(LayoutInflater layoutInflater, String label, View.OnClickListener onClickListener){
        View view = layoutInflater.inflate(R.layout.layout_button, null);

        TextView viewLabel = view.findViewById(R.id.label);
        viewLabel.setText(label);

        view.setOnClickListener(onClickListener);

        return view;
    }

    public static View inflateLockedButton(LayoutInflater layoutInflater, String label){
        View view = layoutInflater.inflate(R.layout.layout_button_locked, null);

        TextView viewLabel = view.findViewById(R.id.label);
        viewLabel.setText(label);

        return view;
    }

    public static View createSpacer(Context context){
        View view = new View(context);
        view.setLayoutParams(new LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                0,
                1.0f
        ));

        return view;
    }

    public static String getIlocanoAudioFileName(String string){
        return "i_" + cleanFileNameString(string);
    }

    public static String getHiligaynonAudioFileName(String string){
        return "h_" + cleanFileNameString(string);
    }

    private static String cleanFileNameString(String string){
        return string.toLowerCase()
                .replace(" ", "_")
                .replace("-", "_")
                .replace("\"", "")
                .replace(",", "_")
                .replace("__", "_");
    }
}

File: app/src/main/java/com/erotc/learning/util/ApplicationUtil.kt
package com.erotc.learning.util

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.widget.EditText
import android.widget.ImageButton
import android.widget.LinearLayout
import android.widget.TextView
import com.erotc.learning.R
import com.erotc.learning.repository.LearnRepository
import com.google.gson.Gson
import java.io.IOException

/**
 * Created on 11/12/2018.
 */
object ApplicationUtil {
    private val LOG_TAG = ApplicationUtil::class.java.simpleName

    const val FILE_TOPIC = "topic.json"
    const val FILE_LECTURE = "lecture.json"
    const val FILE_TUTORIAL = "tutorial.json"
    const val FILE_ASSESSMENT = "assessment.json"

    val gson = Gson()

    fun shouldInitialize(context: Context): Boolean {
        val repository = LearnRepository.getInstance(context)
        return repository.hasMissingData
    }

    fun getStringFromAsset(context: Context, fileName: String): String? {
        val jsonString: String
        try {
            jsonString = context.assets.open(fileName).bufferedReader().use { it.readText() }
        } catch (e: IOException) {
            Log.e(LOG_TAG, Log.getStackTraceString(e))
            return null
        }

        return jsonString
    }

    fun inflateChoice(layoutInflater: LayoutInflater, label: String?, onClickListener: View.OnClickListener?): View {
        val view = layoutInflater.inflate(R.layout.layout_choice, null)
        val viewLabel: TextView = view.findViewById(R.id.label)
        val button:View = view.findViewById(R.id.button)

        viewLabel.text = label
        button.setOnClickListener(onClickListener)

        return view
    }

    fun inflateFill(layoutInflater: LayoutInflater, onValidate: (String?) -> Unit): View {
        val view = layoutInflater.inflate(R.layout.layout_fill, null)
        val input: EditText = view.findViewById(R.id.answer)

        view.findViewById<ImageButton>(R.id.validate).setOnClickListener {
            onValidate(input.text.toString().trim())
        }

        return view
    }

    fun createSpacer(context: Context?): View {
        val view = View(context)
        view.layoutParams = LinearLayout.LayoutParams (
                LinearLayout.LayoutParams.MATCH_PARENT,
                0,
                1.0f
        )

        return view
    }
}